File: MaximumDepthofBinaryTree.cpp
#include <iostream>
#include "Utility.h"
using namespace std;

class Solution
{
public:
    int maxDepth(TreeNode *root)
    {
        if (root == NULL)
            return 0;
        if (root->left == NULL && root->right == NULL)
            return 1;
        int leftd = maxDepth(root->left);
        int rightd = maxDepth(root->right);
        return max(leftd, rightd) + 1;
    }
};

int main()
{
    
    return 0;
}

File: JavaSolution/src/Uber/MaximumDepthofBinaryTree.java
package Uber;

/**
 * Created by xyunpeng on 1/8/16.
 */
public class MaximumDepthofBinaryTree {
    public int maxDepth(TreeNode root) {
        if (root == null) {
            return 0;
        } else {
            return 1 + Math.max(maxDepth(root.left, root.right));
        }
    }
}
